src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/ActionPlanAppointment.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity
+
+import org.hibernate.annotations.Fetch
+import org.hibernate.annotations.FetchMode
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.ManyToOne
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+
+@Entity
+@Table(name = "action_plan_appointment")
+data class ActionPlanAppointment(
+
+  // Attributes
+  @NotNull val sessionNumber: Int,
+
+  // Activities
+  var appointmentTime: OffsetDateTime? = null,
+  var durationInMinutes: Int? = null,
+
+  // Status
+  @NotNull @ManyToOne @Fetch(FetchMode.JOIN) val createdBy: AuthUser,
+  @NotNull val createdAt: OffsetDateTime,
+
+  // Required
+  @NotNull @ManyToOne val actionPlan: ActionPlan,
```

Comment:
do we want to specify the fetchtype here? it's likely we will not want the full action plan when looking up appointments? so lazy?

Reply:
In "maintaining" appointments, I'm not sure we need to eagerly load action plans (and joined activities) as we don't use anything from action plan - so the action plan wont need to get hydrated.